components/dash-core-components/src/components/Tooltip.react.js

```diff
@@ -0,0 +1,276 @@
+import PropTypes from 'prop-types';
+
+import _JSXStyle from 'styled-jsx/style'; // eslint-disable-line no-unused-vars
+
+/**
+ * A tooltip with an absolute position.
+ */
+const Tooltip = props => {
+    const {bbox, border_color, background_color, id, loading_state} = props;
+    const is_loading = loading_state?.is_loading;
+    const show = props.show && bbox;
+
+    return (
+        <>
+            <div className="dcc-tooltip-bounding-box">
+                <span
+                    data-dash-is-loading={is_loading || undefined}
+                    className={`hover hover-${props.direction}`}
+                >
+                    <span
+                        id={id}
+                        className={`hover-content ${props.className}`}
```

Comment:
Discovered while writing the test, but anyway users will expect `props.id` to be in the DOM. I put it on the content `<span>`, because this is where `props.className` is applied, and that's mostly where you probably want your styles applied. Makes it a little awkward to style the container, but because of the pseudo-elements making the triangle this is discouraged anyway, and we have special props for bg and border colors there already.